app/app.ts

```diff
@@ -86,10 +93,40 @@ export class DoseAmigosApp {
                 // schedule an initial token refresh for when the
                 // token expires
                 this.auth.startupTokenRefresh();
+
+                // Notifications are scheduled/rescheduled every 5 minutes
+                this.nIntervId = setInterval(this.scheduleNotifications, 300000);
+
             }
         );
     }
 
+    private scheduleNotifications() {
+
+        // if user is logged in, clear all notifications
+        // get a current list of doseEvents, and
+        // schedule new notifications
+        if (this.auth.authenticated()) {
+
+            // Clear all previous notifications.
+            this.notification.clearAll();
```

Comment:
1. Wrong method:
Clear means remove the history of notifications that already fired.
Cancel means remove it if it already fired (clear) AND do not fire ones that haven't happened yet.
This should be `cancelAll()`.

https://github.com/katzer/cordova-plugin-local-notifications/wiki/06.-Clear
https://github.com/katzer/cordova-plugin-local-notifications/wiki/07.-Cancelation

2. We shouldn't cancel notifications until we successfully get the DoseEvents from the server.  What if we were offline, or the call failed? You just canceled the notifications, and didn't replace them.